pkg/virtualgarden/kube_api_server_deployments.go

```diff
@@ -1,9 +1,609 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package virtualgarden
 
+import (
+	"context"
+	_ "embed"
+	"fmt"
+
+	"github.com/gardener/gardener/pkg/utils"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	KubeAPIServerDeploymentNameAPIServer         = Prefix + "-kube-apiserver"
+	KubeAPIServerDeploymentNameControllerManager = Prefix + "-kube-controller-manager"
+)
+
+func (o *operation) deployDeployments(ctx context.Context, checksums map[string]string, basicAuthPw string) error {
+	o.log.Infof("Deploying deployments for the kube-apiserver")
+
+	if err := o.deployKubeAPIServerDeployment(ctx, checksums, basicAuthPw); err != nil {
+		return err
+	}
+
+	if err := waitForDeploymentReady(ctx, o.client, o.emptyDeployment(KubeAPIServerDeploymentNameAPIServer)); err != nil {
+		return err
+	}
+
+	if err := o.deployKubeAPIServerDeploymentControllerManager(ctx, checksums, basicAuthPw); err != nil {
+		return err
+	}
+
+	if err := waitForDeploymentReady(ctx, o.client, o.emptyDeployment(KubeAPIServerDeploymentNameControllerManager)); err != nil {
+		return err
+	}
```

Comment:
Unfortunately, I do not understand what you mean by that.

Reply:
Do you mean we should use two separate steps in the flow graph?

Reply:
Yep.

Reply:
we changed this such that the flow is more obvious but we did not make 4 flow graph steps for it. This would artificially split the deployment of the api server logic and data flow between different step has to be implemented making the coding less readable (e.g. basicAuthCredentials).